docs/development-guidelines.md

```diff
@@ -0,0 +1,75 @@
+# Development Guidelines
+
+## Quick Start
+
+Check whether something [Needs To Be Installed](#needs-to-be-installed).
+
+### Building and Provenances
+
+To build a binary from the Git repository specified in [`testdata/build.toml`](../testdata/build.toml) and generate its provenance file, run either:
+
+```bash
+$ bazel run  //cmd/build:main -- \
+  -build_config_path <absolute-path-to-transparent-release-repo>/testdata/build.toml \
+```
+
+or, alternatively:
+
+```bash
+$ go run cmd/build/main.go -build_config_path testdata/build.toml
+```
+
+You should see the following output on the console:
+
+```bash
+2022/04/14 09:08:17 The hash of the binary is: 15dc16c42a4ac9ed77f337a4a3065a63e444c29c18c8cf69d6a6b4ae678dca5c
+2022/04/14 09:08:17 Storing the provenance in <your-path>/transparent-release/provenance.json
+```
+
+To build from a local repository you can specify `-git_root_dir`. In this case, the binary will be built from the repo, only if the latest commit matches the one specified in the config file:
```

Comment:
Maybe add that "otherwise the command will fail with an error".

Reply:
Done.